### PR TITLE
feat(style): implement `From` trait for crossterm to `Style` related structs

### PR DESCRIPTION
**Pull request by Valentin271**
This PR includes the following `From` implementations:

- from `Attribute` to `Modifier`
- from `Attributes` to `Modifier`
- from crossterm `Color` to `Color`
- from `ContentStyle` to `Style`

This PR relates to #679. I've decided to split the work across multiple PRs to facilitate the reviewing process.

### CODE REVIEW

File: src/style.rs [hunk 1]
```diff
@@ -110,6 +110,60 @@ impl fmt::Debug for Modifier {
     }
 }
 
+#[cfg(feature = "crossterm")]
```
Comment (joshka) [inline src/style.rs:113]: These might belong in crossterm.rs to avoid the feature checks and having an impl for each backend in style.rs

File: src/style.rs [hunk 2]
```diff
@@ -110,6 +110,60 @@ impl fmt::Debug for Modifier {
...
+            res |= Modifier::DIM;
+        }
+        if value.has(CAttribute::Italic) {
+            res |= Modifier::ITALIC;
+        }
+        if value.has(CAttribute::Underlined)
+            || value.has(CAttribute::DoubleUnderlined)
+            || value.has(CAttribute::Undercurled)
+            || value.has(CAttribute::Underdotted)
+            || value.has(CAttribute::Underdashed)
```
Comment (joshka) [inline src/style.rs:143]: Aside: If these really work, do you think we should add them to Ratatui at some point?

File: src/style.rs [hunk 3]
```diff
@@ -631,4 +826,67 @@ mod tests {
...
+            ContentStyle {
+                attributes: CAttributes::from([CAttribute::NoBold, CAttribute::NoItalic].as_ref()),
+                ..Default::default()
+            },
+        ];
+        let styles = styles();
+
+        for (content_style, style) in content_styles.into_iter().zip(styles) {
+            assert_eq!(Style::from(content_style), style);
+        }
```
Comment (joshka) [inline src/style.rs:876]: When reading it I had to read down, then up to work out what fn styles was doing and then for each style had to read up and down to keep them in sync. This is a test smell. I'd prefer to replace this approach with many asserts like:
```rust
assert_eq!(ContentStyle::new().dark_yellow().into(), Style::new().yellow());
```

**Review by joshka — CHANGES_REQUESTED** [review]

File: src/style.rs (hunk 1 above)
Comment (Valentin271) [reply src/style.rs]: Makes much more sense now that I'm writing it, to have it in crossterm with the other From impl

File: src/style.rs (hunk 3 above)
Comment (Valentin271) [reply src/style.rs]: You're right I also had to do that while writing the test. Fixed when moved to crossterm

File: src/style.rs (hunk 2 above)
Comment (Valentin271) [reply src/style.rs]: I think we either implement everything (see [crossterm](https://docs.rs/crossterm/latest/crossterm/style/enum.Attribute.html) and [ANSI wiki](https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_(Select_Graphic_Rendition)_parameters)) or stay as is.

It seems those are rarely supported features. For example my `xfce4-terminal` supports underlined and double underlined bot not fraktur or framed. And tbh `Underdotted`, `Undercurled` and `Underdashed` doesn't even show up on the wiki.

tldr; overall no one has requested that, let's keep it as is and implement it later if really needed

File: src/backend/crossterm.rs [hunk 4]
```diff
@@ -274,6 +274,32 @@ impl From<Color> for CColor {
     }
 }
 
+impl From<CColor> for Color {
+    fn from(value: CColor) -> Self {
+        match value {
+            CColor::Reset => Self::Reset,
```
Comment (joshka) [inline src/backend/crossterm.rs:280]: nit: is it worth making these in the same order as above (it's that way because it aligns with the numeric index of the colors)?

File: src/backend/crossterm.rs [hunk 5]
```diff
@@ -344,3 +370,300 @@ impl ModifierDiff {
         Ok(())
     }
 }
+
+impl From<CAttribute> for Modifier {
+    fn from(value: CAttribute) -> Self {
+        Modifier::from(CAttributes::from(value))
```
Comment (joshka) [inline src/backend/crossterm.rs:376]: Can you add a doc comment for this line - it's a bit weird / requires understanding of crossterm Attribute vs Attributes

**Review by joshka — APPROVED** [review]
Approve - with nits

File: src/backend/crossterm.rs (hunk 4 above)
Comment (Valentin271) [reply src/backend/crossterm.rs:280]: Sure, I put them in the order of the enum originally

File: src/backend/crossterm.rs (hunk 5 above)
Comment (Valentin271) [reply src/backend/crossterm.rs:376]: Sure, plus the difference is subtle

**Review by orhun — APPROVED** [review]
I likey
